app/models/concerns/issue_search.rb

```diff
@@ -0,0 +1,189 @@
+module IssueSearch
+  extend ActiveSupport::Concern
+
+  included do
+    include Elasticsearch::Model
+    include Elasticsearch::Model::Callbacks
+
+
+    FIELDS = ['full_title^2', 'exact_title^2', 'body']
+
+    settings index: { number_of_shards: 1, number_of_replicas: 0 } do
+      mapping do
+        indexes :full_title, :analyzer => 'snowball', :boost => 6
+        indexes :exact_title, :index => :not_analyzed, :boost => 2
+
+        indexes :created_at, type: 'date'
+        indexes :updated_at, type: 'date'
+        indexes :closed_at, type: 'date'
+
+        indexes :owner_id, type: 'integer'
+        indexes :stargazers_count, type: 'integer'
+        indexes :github_id, type: 'integer'
+        indexes :github_contributions_count, type: 'integer'
+
+        indexes :number
+        indexes :state
+
+      end
+    end
+
+    after_save() { __elasticsearch__.index_document }
+
+    def as_indexed_json(options = {})
+      as_json methods: [:exact_title]
```

Comment:
Need to add in `github_repository_language` and `github_repository_license` here and add the same named methods to `GitHubIssue`